mlflow/utils/autologging/__init__.py

```diff
@@ -618,7 +619,7 @@ def call(cls, original, *args, **kwargs):
     def __call__(self, original, *args, **kwargs):
         try:
             return self._patch_implementation(original, *args, **kwargs)
-        except Exception as e:
+        except (Exception, KeyboardInterrupt) as e:
```

Comment:
Instead of handling `KeyboardInterrupt`, I also considered handling SIGINTS via `signal.signal` (https://docs.python.org/3/library/signal.html#signal.signal). However, performing cleanup logic in `signal.signal` has some scary implications (e.g., it can lead to deadlock if the cleanup operating makes use of `logger` since `logger` uses a non-reetrant lock & a signal handler may be called in the middle of another logging statement).

While `KeyboardInterrupt` isn't guaranteed to fire 100% of the time on a SIGINT, it should be "good enough" for most practical cases.